components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -410,8 +411,12 @@ private void updateBasicApplicationData(ServiceProvider serviceProvider, Connect
 
         // only if the application has been renamed
         if (!StringUtils.equals(applicationName, storedAppName)) {
+            String applicationNameforRole = ApplicationConstants.APPLICATION_DOMAIN + UserCoreConstants.DOMAIN_SEPARATOR
+                    + applicationName;
+            String storedAppNameforRole = ApplicationConstants.APPLICATION_DOMAIN + UserCoreConstants.DOMAIN_SEPARATOR
+                    + storedAppName;
```

Comment:
When we use this method it will append the domain name with uppercase. In that case the UM_HYBRID_ROLE will not updated as it searches for lowercase domain name when updating.